### PR TITLE
Add "Noduino Quantum" board

### PR DESCRIPTION
**Pull request by icamgo**
The information of Noduino Quantum board: http://wiki.jackslab.org/Noduino_Quantum

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -201,3 +201,42 @@ espea32.menu.UploadSpeed.460800.upload.speed=460800
...
+quantum.upload.maximum_data_size=294912
+quantum.upload.wait_for_upload_port=true
+
+quantum.serial.disableDTR=true
+quantum.serial.disableRTS=true
+
+quantum.build.mcu=esp32
+quantum.build.core=esp32
+quantum.build.variant=quantum
+quantum.build.board=Quantum
```
Comment (me-no-dev) [inline boards.txt]: can you please change Quantum to all upper case?

Comment (comcat) [inline boards.txt]: like this?
quantum.build.board=QUANTUM

Comment (me-no-dev) [inline boards.txt]: yes :)

**Comment by me-no-dev** [comment]
thanks will merge in a bit :) I have some other changes lined up

**Pull request merged**